pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -553,6 +567,43 @@ func applyManagementResources(
 	return currentGenerations, appliedErrs, nil
 }
 
+// updateStoredVersion update(remove) deleted api version from CRD status.StoredVersions
+func updateStoredVersion(ctx context.Context, apiExtensionClient apiextensionsclient.Interface) error {
+	for name, version := range removedCRDStoredVersions {
+		// retrieve CRD
+		crd, err := apiExtensionClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, name, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			continue
+		}
+		if err != nil {
+			klog.Warningf("faield to get CRD %v: %v", crd.Name, err)
+			continue
+		}
+
+		// remove v1alpha1 from its status
+		oldStoredVersions := crd.Status.StoredVersions
+		newStoredVersions := make([]string, 0, len(oldStoredVersions))
+		for _, stored := range oldStoredVersions {
+			if stored != version {
+				newStoredVersions = append(newStoredVersions, stored)
+			}
+		}
+
+		if !reflect.DeepEqual(oldStoredVersions, newStoredVersions) {
+			crd.Status.StoredVersions = newStoredVersions
+			// update the status sub-resource
+			crd, err = apiExtensionClient.ApiextensionsV1().CustomResourceDefinitions().UpdateStatus(ctx, crd, metav1.UpdateOptions{})
+			if err != nil {
+				klog.Warningf("faield to update CRD %v: %v", crd.Name, err)
+				continue
```

Comment:
It's not enough to output warning here. Instead, an error should be returned in this case, because the update of the crd (removing v1alpha1) will fail if v1alpha1 is still in the store version list.

Reply:
fixed